### DISCUSSION TITLE
Playlists management like in other players

### DISCUSSION THREAD

**Discussion by Solarunit** [MusicPlayerDaemon/MPD#1843] [General]
The biggest issue I have with mpd (not sure if it's client app or server) is the Playlists management like other players have:
- you create a new Playlist and add any folders/songs in it
- you can select any Playlist from the list, click on the song, it starts playing 
- select other playlist and play any song immediately

mpd doesn't work like that, instead you should "activate" playlist first (it starts playing the 1st song) and then select a song from this playlist. It looks like mpd only has a single active playlist mode where you can add/remove songs, clear it etc.. If you "activate" other playlist it becomes an active playlist.

Is there a way to configure mpd to have independent playlists like other players have?

**Comment by MaxKellermann** [comment]
I don't get your point. With MPD, you can
- create a new Playlist and add any folders/songs in it
- load any Playlist from the list, and start playing
- load other playlist and play any song immediately

> instead you should "activate" playlist first

MPD has a queue of songs that shall be played. You can load any playlist into the queue, or load multiple playlists.

> it starts playing the 1st song

No, that's just one special case. You can select any song from the queue and play it.

> It looks like mpd only has a single active playlist mode where you can add/remove songs, clear it etc..

MPD has the queue, but why does that bother you? You can do all the things you said.

> Is there a way to configure mpd to have independent playlists like other players have?

MPD does have many independent playlists.

MPD is just a low-level music player without a user interface. It can do all the things you want it to do. I believe the things that bother you is just how MPD's features are shown in the MPD clients you're using. I believe you want a MPD client that behaves differently.

**Comment by Solarunit** [comment]
>MPD has a queue of songs that shall be played. You can load any playlist into the queue, or load multiple playlists.

Right, I named this queue of songs as "single active playlist mode".  I will try to explain how it works in other players with heavy focus on playlists:

- you create many playlists, you don't need a queue of songs, you just select any playlist, double click on a song to start playing, and player would start playing that playlist until the last song or indefinitely (depending on the setting). If you need to switch to other playlist-2, just select to play any song from from playlist-2 and so on..

How it works in ncmpcpp:

- a queue of songs is the 1 window called Main Playlist.
- it has window 5 "Playlist editor" but it's just an editor (unfortunately), if you select playlist-1 and start playing the 1st song, that song would be added to the Main Playlist (window 1) to the end of it.  Or you can add the whole playlist-1 to the Main Playlist (window 1) but you probably need to clear the Main Playlist first because it could contain the previously added songs from another playlists.

So in order to reproduce the workflow of other players I have to constantly clear the Main Playlist in ncmpcpp and load my saved playlist into it. If you constantly switching between many playlist it would become pretty annoying I think.
I was expected that I could just use window 5 "Playlist editor" to select playlist and play songs without the need to use Main Playlist at all. But it doesn't work this way.

Maybe I am using ncmpcpp wrong?

I hope I described it clear :)

>It can do all the things you want it to do. I believe the things that bother you is just how MPD's features are shown in the MPD clients you're using. I believe you want a MPD client that behaves differently.

Yes, well, I am not sure maybe it's MPD clients and their quirks.. I am using ncmpcpp and MALP and they both work not the way I want: single click/tap select playlist from the list of saved playlists and single click/tap to select a song from selected playlist to play.

**Comment by Emilot** [comment]
@Solarunit take a look at this project:

https://github.com/jcorporation/myMPD

I think you will find what you are looking...

**Reply by Solarunit** [reply]
I see it's web-based MPD client, and I am not into web-based apps, but I will check it out. I see there is Docker Compose example in the docks, nice :)

**Reply by MaxKellermann** [reply]
If you're into console UIs like me, why don't you help improving ncmpc so it suits your needs better?
(Or if you, for some reason, prefer ncmpcpp over ncmpc - the ncmpcpp project would sure welcome help, too)

**Reply by Solarunit** [reply]
I haven't tried ncmpc yet. Well, I can only help with feature requests, maybe I will add some :)

**Comment by Solarunit** [comment]
Ok, I installed MyMPD and played with it a bit. It works like others mpd client:

- there is a Main playlist (queue) were you load your saved playlist
- single click appends saved playlist to the queue (this is opposite of what I want)
- there is option to Replace queue and Play for the saved playlist, again it relies on queue. 
____


I think the client should have an option where **the main queue = activated playlist**,  you can't add/delete songs to that queue. When you select a new saved playlist the new instance of queue is created. So each playlist creates their own queue.

Something like this (when you deal with playlists)

**Reply by jcorporation** [reply]
myMPD maintainer here. My opinion is, that MPD does not work in the way you want. MPD has only a single queue from which you can play songs. Each playlist must be added to this queue before you can play a song. You simply can not play a song directly without adding it to the queue. That is not a client limitation, MPD simply works that way.

**Reply by Solarunit** [reply]
Thanks for the reply! That was my initial assumption. I even created a whole Feature Request for ncmpcpp https://github.com/ncmpcpp/ncmpcpp/issues/578

I guess my only hope if this feature would be considered for mpd server.
I saw a couple of similar questions when I googled a solution for my problem.

**Reply by kingosticks** [reply]
> myMPD maintainer here. My opinion is, that MPD does not work in the way you want. MPD has only a single queue from which you can play songs. Each playlist must be added to this queue before you can play a song. You simply can not play a song directly without adding it to the queue. That is not a client limitation, MPD simply works that way.

I don't follow. Yes, the client has to add the song to mpd's queue before playing but the client doesn't have to expose this to the user. I don't understand why the client can't hide this implementation detail and provide the requested "one-click" user interface.

**Reply by jcorporation** [reply]
I think, the queue is the main design concept around all other features are implemented, not only a detail. Isn’t it?

**Reply by kingosticks** [reply]
There's no requirement to expose it to the user in every interaction, that's just the easiest thing to do.

**Comment by Solarunit** [comment]
I created a Feature Requests for ncmpcpp
https://github.com/ncmpcpp/ncmpcpp/issues/578

The idea is this:
>a separate, independent "Main Playlist/Main queue" instance for each saved Playlists. So you don't need to clear Main Playlist anymore before loading another saved playlist, instead you just select saved playlist and click a song to play, that automatically would activate a separate queue associated only with that particular saved playlist.


I think every mpd client should have this (or option at least)..
